src/engine/CodeValidator.v3

```diff
@@ -1087,6 +1087,53 @@ class CodeValidator(extensions: Extension.set, limits: Limits, module: Module, e
 				I64_ATOMIC_RMW_CMPXCHG	=> check_MEMARG(opcode, 8);
 				I32_ATOMIC_RMW8_CMPXCHG_U	=> check_MEMARG(opcode, 1);
 				I32_ATOMIC_RMW16_CMPXCHG_U	=> check_MEMARG(opcode, 2);
+				CONT_NEW => {
+					var cont_decl = parser.readCont();
+					popE(ValueTypes.Ref(true, cont_decl.sig));
+					// TODO[ss]: nullable = false as per proposal doc. Confirm/ask this.
+					val_stack.push(ValueTypes.Ref(false, cont_decl));
+				}
+				CONT_BIND => {
+					var cont_a = parser.readCont();
+					popE(ValueTypes.Ref(true, cont_a));
+					var cont_b = parser.readCont();
+
+					var sig_a = cont_a.sig;
+					var sig_b = cont_b.sig;
+
+					var source_length = sig_a.params.length;
+					var target_length = sig_b.params.length;
+					if (source_length < target_length) {
+						err_atpc().IllegalContinuationBind(cont_a, cont_b);
+					}
+
+					var n_binds = source_length - target_length;
+					var applied_args = Arrays.range(sig_a.params, 0, n_binds);
+					var remain_args = Arrays.range(sig_a.params, n_binds, source_length);
+
+					// Resulting signature after partial application matches the given signature `sig_b`.
+					// TODO[ss]: Make sure subtyping is checked correctly here.
+					var result_sig = SigDecl.new(true, ValueTypes.NO_HEAPTYPES, remain_args, sig_a.results);
+					if (!result_sig.isAssignableSig(sig_b)) {
```

Comment:
We should maybe factor out a different routine that doesn't work on sigs, but parameters and returns (or maybe just an Array<ValueType>, because the `isAssignableSig` is a hold-over from a more general structural type system that was not adopted in Wasm.